PolyEngine/Standalone/Src/Main.cpp

```diff
@@ -101,15 +100,39 @@ int main(int argc, char* args[])
 				Engine->UpdateMousePos(Poly::Vector2i(event.motion.x, event.motion.y));
 				break;
 			case SDL_MOUSEBUTTONDOWN:
-                Engine->MouseButtonDown(static_cast<Poly::eMouseButton>(event.button.button));
-                break;
+				Engine->MouseButtonDown(static_cast<Poly::eMouseButton>(event.button.button));
+				break;
 			case SDL_MOUSEBUTTONUP:
-                Engine->MouseButtonUp(static_cast<Poly::eMouseButton>(event.button.button));
+				Engine->MouseButtonUp(static_cast<Poly::eMouseButton>(event.button.button));
 				break;
 			case SDL_MOUSEWHEEL:
 				// Not sure if this is correct.
 				Engine->UpdateWheelPos(Poly::Vector2i(event.wheel.x, event.wheel.y));
 				break;
+			case SDL_CONTROLLERBUTTONDOWN:
+				Engine->ControllerButtonDown(event.cbutton.which, static_cast<Poly::eControllerButton>(event.cbutton.button));
+				break;
+			case SDL_CONTROLLERBUTTONUP:
+				Engine->ControllerButtonUp(event.cbutton.which, static_cast<Poly::eControllerButton>(event.cbutton.button));
+				break;
+			case SDL_CONTROLLERAXISMOTION:
+				Engine->ControllerAxisMotion(event.caxis.which, static_cast<Poly::eControllerAxis>(event.caxis.axis), event.caxis.value);
+				break;
+			case SDL_CONTROLLERDEVICEADDED:
+			{
+				SDL_GameController* controller = SDL_GameControllerOpen(event.cdevice.which);
+				SDL_Joystick* joystickPtr = SDL_GameControllerGetJoystick(controller);
+				i32 joystickID = SDL_JoystickInstanceID(joystickPtr);
+				Engine->AddController(joystickID);
+				break;
+			}
+			case SDL_CONTROLLERDEVICEREMOVED:
+			{
+				SDL_GameController* controller = SDL_GameControllerFromInstanceID(event.cdevice.which);
+				SDL_GameControllerClose(controller);
+				Engine->RemoveController(event.cdevice.which);
```

Comment:
It would be better to first remove the controller (from engine) and then close it.